src/model_metric_reporter.cpp

```diff
@@ -16,138 +16,169 @@
-        {"protocol", "kserve"},
-        {"method", "modelinfer"},
-        {"interface", "grpc"}});
-    this->requestFailGrpcModelMetadata = family->addMetric({{"name", modelName},
-        {"version", std::to_string(modelVersion)},
-        {"protocol", "kserve"},
-        {"method", "modelmetadata"},
-        {"interface", "grpc"}});
-    this->requestFailGrpcModelStatus = family->addMetric({{"name", modelName},
-        {"version", std::to_string(modelVersion)},
-        {"protocol", "kserve"},
-        {"method", "modelstatus"},
-        {"interface", "grpc"}});
-    this->requestFailRestModelInfer = family->addMetric({{"name", modelName},
-        {"version", std::to_string(modelVersion)},
-        {"protocol", "kserve"},
-        {"method", "modelinfer"},
-        {"interface", "rest"}});
-    this->requestFailRestModelMetadata = family->addMetric({{"name", modelName},
-        {"version", std::to_string(modelVersion)},
-        {"protocol", "kserve"},
-        {"method", "modelmetadata"},
-        {"interface", "rest"}});
-    this->requestFailRestModelStatus = family->addMetric({{"name", modelName},
-        {"version", std::to_string(modelVersion)},
-        {"protocol", "kserve"},
-        {"method", "modelstatus"},
-        {"interface", "rest"}});
+    if (metricConfig->requestFailGrpcModelInfer)
+        this->requestFailGrpcModelInfer = family->addMetric({{"name", modelName},
+            {"version", std::to_string(modelVersion)},
+            {"protocol", "kserve"},
+            {"method", "modelinfer"},
+            {"interface", "grpc"}});
+    if (metricConfig->requestFailGrpcModelMetadata)
+        this->requestFailGrpcModelMetadata = family->addMetric({{"name", modelName},
+            {"version", std::to_string(modelVersion)},
+            {"protocol", "kserve"},
+            {"method", "modelmetadata"},
+            {"interface", "grpc"}});
+    if (metricConfig->requestFailGrpcModelStatus)
+        this->requestFailGrpcModelStatus = family->addMetric({{"name", modelName},
+            {"version", std::to_string(modelVersion)},
+            {"protocol", "kserve"},
+            {"method", "modelstatus"},
+            {"interface", "grpc"}});
+    if (metricConfig->requestFailRestModelInfer)
+        this->requestFailRestModelInfer = family->addMetric({{"name", modelName},
+            {"version", std::to_string(modelVersion)},
+            {"protocol", "kserve"},
+            {"method", "modelinfer"},
+            {"interface", "rest"}});
+    if (metricConfig->requestFailRestModelMetadata)
+        this->requestFailRestModelMetadata = family->addMetric({{"name", modelName},
+            {"version", std::to_string(modelVersion)},
+            {"protocol", "kserve"},
+            {"method", "modelmetadata"},
+            {"interface", "rest"}});
+    if (metricConfig->requestFailRestModelStatus)
+        this->requestFailRestModelStatus = family->addMetric({{"name", modelName},
```

Comment:
There is no ModelStatus in KServer - there is ModelReady instead

Reply:
Fixed.